neuralforecast/core.py

```diff
@@ -380,9 +384,33 @@ def _prepare_fit_distributed(
             min_size=df.groupBy(id_col).count().agg({"count": "min"}).first()[0],
         )
 
+    def _prepare_iterative_fit(
+        self, df, static_df, sort_df, id_col, time_col, target_col
+    ):
+        if self.local_scaler_type is not None:
+            raise ValueError(
+                "Historic scaling isn't supported in distributed. "
+                "Please open an issue if this would be valuable to you."
+            )
+
+        self.id_col = id_col
+        self.time_col = time_col
+        self.target_col = target_col
+        self.scalers_ = {}
+        self.sort_df = sort_df
+
+        return IterativeTimeSeriesDataset.from_data_directory(
+            files=df,
+            static_df=static_df,
+            sort_df=sort_df,
+            id_col=id_col,
+            time_col=time_col,
+            target_col=target_col,
+        )
+
     def fit(
         self,
-        df: Optional[Union[DataFrame, SparkDataFrame]] = None,
+        df: Optional[Union[DataFrame, SparkDataFrame, List[str]]] = None,
```

Comment:
having the arg name as `df` isn't ideal really. I wonder if we could add an optional arg for files_list or something